feat/FECA-44 Address Details - Add address feature

rajeh1032 opened the pull request:
- Introduced AddressApiServices for API interactions.
- Implemented AddressLocalDataSource for local data handling.
- Created models for Address, City, and Governorate with JSON serialization.
- Developed repository pattern for address management.
- Added use cases for CRUD operations on addresses.
- Integrated Google Maps and location permissions in iOS.
- Added governorate data in JSON format for local use.

<!--
  🛠️ For Work In Progress, please use Draft PRs:
  https://github.blog/2019-02-14-introducing-draft-pull-requests/

  ✅ Before submitting:
  - Follow your project's contributing guide
  - Ensure code compiles and passes all tests
  - Use descriptive commit messages
  - Keep PRs small and focused
-->

## 📌 What type of PR is this? (check all applicable)

- [ ✅] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🧹 Refactor
- [ ] 🚀 Performance Optimization
- [ ] 🧾 Documentation Update
- [ ] ♿ Accessibility Fix
- [ ] 🔧 CI/CD or Config Change

---

## 📝 Description


Added **Address Feature** 

---

## 🧩 Related Issues / Tickets

- Closes #123

---

## 🧪 QA Instructions

_How to test the changes manually?_

- [ ✅ ] Run the app on Android
- [ ] Run the app on iOS

- [ ] Go to the products page
- [ ] Try searching for a product (e.g. "Rose")
- [ ] Verify that results are shown correctly
- [ ] Verify "No products found" message when nothing matches


---

## 📸 Screenshots or Recordings (if UI changes)
<img width="613" height="1368" alt="image" src="https://github.com/user-attachments/assets/01f1d396-1df6-408b-847d-4942696a15d0" />
<img width="602" height="1370" alt="image" src="https://github.com/user-attachments/assets/8b0fba58-21c7-4171-9ad1-d591592039fd" />
<img width="616" height="1381" alt="image" src="https://github.com/user-attachments/assets/0cd17bf1-1436-4d6e-88ec-2c360a250877" />



_Add screenshots or screen recordings to demonstrate your changes._
*(Optional – e.g., showing search results and empty state)*



---

## ♿ UI Accessibility Checklist (if applicable)

- [ ] Semantic widget structure used (`Semantics`, `Accessible`, etc.)
- [ ] Supports screen readers
- [ ] Good color contrast
- [ ] Tested with TalkBack/VoiceOver

---

## 🧪 Tests

- [ ✅] Unit tests added for search feature
- [ ] Widget tests added for search UI
- [ ] Integration tests added
- [ ] No tests needed (explain why):

---

## 🛠 Post-Deployment Tasks (Optional)




_No extra tasks needed._


---

## 🎬 Fun GIF (Optional)

![fun gif](https://media.giphy.com/media/xT0xeJpnrWC4XWblEk/giphy.gif)

rajeh1032 committed:
feat: Implement Sign Up feature with form validation and API integration

- Added SignUpState to manage loading, success, and error states.
- Created UI components for sign-up including email, password, name, phone, and gender fields.
- Implemented BuildElevatedButton for handling loading states.
- Developed SignUpForm to manage user input and submission.
- Integrated API service for sign-up functionality with error handling.
- Added localization support for various UI texts.
- Created SecureStorage for managing secure data storage.
- Implemented unit tests for AuthRemoteDataSource to ensure API integration works as expected.

rajeh1032 committed:
feat: Refactor Sign Up feature by restructuring view model and updating imports

rajeh1032 committed:
Refactor : Refactor sign-up flow and improve user data handling

- Added generated files to .gitignore for cleaner repository.
- Removed unused translations from Arabic JSON file.
- Updated English translations to include success message for account creation.
- Refactored AuthRemoteDataSourceImpl to use final ApiServices and improved error handling.
- Enhanced SignUpResponseDtoMapper to handle null values and provide default UserEntity.
- Updated UserDtoMapper to ensure all fields are nullable with default values.
- Deleted obsolete sign_in_response_dto.g.dart file.
- Refactored SignUpResponseDto to use nullable fields and updated JSON serialization.
- Cleaned up SignUpRequestDto by removing unnecessary comments.
- Updated UserDto to make fields nullable and added JSON annotations.
- Removed BuildElevatedButton widget as it was no longer needed.
- Updated BuildEmailField, BuildFirstAndLastNameField, BuildGenderField, BuildNavigationText, BuildPasswordAndConfirmField, and BuildPhoneField to use Theme for styling.
- Refactored SignUpForm to use new padding constants and removed unnecessary widgets.
- Updated dependency injection configuration to include SignupViewModel.
- Improved CustomBackButton padding using constants.
- Added gender constants to AppConstants for better maintainability.
- Introduced new size constants for consistent padding and margins across the app.

rajeh1032 committed:
feat: Update sign-up fields layout for improved readability and consistency

rajeh1032 committed:
refactor: Refactor sign-up request model and update localization keys

- Replaced SignUpRequestDto with SignUpRequestModel in API service and related files.
- Updated localization keys for gender labels, success messages, and form field labels.
- Enhanced the sign-up form to utilize new localization keys for better internationalization support.
- Removed deprecated SignUpRequestDto and its generated files.
- Improved error handling in ServerFailure class to extract error messages more effectively.
- Added new spacing constants for better layout management in the sign-up form.

rajeh1032 committed:
feat: Add selectedGender to SignUpState and update related methods

rajeh1032 committed:
feat: Update selectedGender to use AppConstants and refactor SignUpForm to StatelessWidget

rajeh1032 committed:
feat: implement profile edit functionality with comprehensive testing

- Add ProfileViewModel with BLoC pattern for profile management
- Implement complete profile API layer with edit, get user data, and photo upload endpoints
- Create comprehensive unit tests for data sources, repositories, and view models
- Add profile-related domain entities, use cases, and mappers
- Update CI/CD workflows to include code generation and localization steps
- Add image picker dependency and profile edit screen routing
- Include Arabic and English translations for profile edit UI
- Fix phone number validation regex to support Egyptian formats

rajeh1032 committed:
feat: implement profile edit functionality with comprehensive testing

- Add ProfileViewModel with BLoC pattern for profile management
- Implement complete profile API layer with edit, get user data, and photo upload endpoints
- Create comprehensive unit tests for data sources, repositories, and view models
- Add profile-related domain entities, use cases, and mappers
- Update CI/CD workflows to include code generation and localization steps
- Add image picker dependency and profile edit screen routing
- Include Arabic and English translations for profile edit UI
- Fix phone number validation regex to support Egyptian formats

rajeh1032 committed:
feat: implement profile edit functionality with comprehensive testing

- Add ProfileViewModel with BLoC pattern for profile management
- Implement complete profile API layer with edit, get user data, and photo upload endpoints
- Create comprehensive unit tests for data sources, repositories, and view models
- Add profile-related domain entities, use cases, and mappers
- Update CI/CD workflows to include code generation and localization steps
- Add image picker dependency and profile edit screen routing
- Include Arabic and English translations for profile edit UI
- Fix phone number validation regex to support Egyptian formats

rajeh1032 committed:
feat: implement profile edit functionality with comprehensive testing

- Add ProfileViewModel with BLoC pattern for profile management
- Implement complete profile API layer with edit, get user data, and photo upload endpoints
- Create comprehensive unit tests for data sources, repositories, and view models
- Add profile-related domain entities, use cases, and mappers
- Update CI/CD workflows to include code generation and localization steps
- Add image picker dependency and profile edit screen routing
- Include Arabic and English translations for profile edit UI
- Fix phone number validation regex to support Egyptian formats

rajeh1032 committed:
feat: implement edit profile screen with image picker and form validation

rajeh1032 committed:
feat: implement edit profile screen with image picker and form validation

rajeh1032 committed:
feat: implement edit profile screen with image picker and form validation

rajeh1032 committed:
feat: implement edit profile screen with image picker and form validation

rajeh1032 committed:
fix pubspec

rajeh1032 committed:
fix pubspec

rajeh1032 committed:
fix pubspec

rajeh1032 committed:
fix pubspec

rajeh1032 committed:
feat: Add address management feature with local and remote data sources

- Introduced AddressApiServices for API interactions.
- Implemented AddressLocalDataSource for local data handling.
- Created models for Address, City, and Governorate with JSON serialization.
- Developed repository pattern for address management.
- Added use cases for CRUD operations on addresses.
- Integrated Google Maps and location permissions in iOS.
- Added governorate data in JSON format for local use.

rajeh1032 committed:
feat: Add address management feature with local and remote data sources

- Introduced AddressApiServices for API interactions.
- Implemented AddressLocalDataSource for local data handling.
- Created models for Address, City, and Governorate with JSON serialization.
- Developed repository pattern for address management.
- Added use cases for CRUD operations on addresses.
- Integrated Google Maps and location permissions in iOS.
- Added governorate data in JSON format for local use.

rajeh1032 committed:
feat: add address management feature with Google Maps integration

- Add comprehensive address CRUD operations with API integration
- Implement Google Maps widget for location selection with current location support
- Create address form with governorate/city dropdowns and validation
- Add location card component for displaying saved addresses
- Extend app routing and navigation for address screens
- Update app sizes constants for new UI components
- Add localization support for address-related strings
- Include comprehensive test coverage for data sources and repositories

rajeh1032 committed:
feat: add address management feature with Google Maps integration

- Add comprehensive address CRUD operations with API integration
- Implement Google Maps widget for location selection with current location support
- Create address form with governorate/city dropdowns and validation
- Add location card component for displaying saved addresses
- Extend app routing and navigation for address screens
- Update app sizes constants for new UI components
- Add localization support for address-related strings
- Include comprehensive test coverage for data sources and repositories

rajeh1032 committed:
Merge branch 'sign-up' into development

rajeh1032 committed:
feat: add sign up endpoint and update related imports

rajeh1032 committed:
Merge branch 'development' of https://github.com/Eng-M0ataz/Flower-E-Commerce-App into development

john-safwat reviewed (commented):
please resolve conflicts

rajeh1032 committed:
Merge branch 'development' into Feature/Address-Details/FECA-44

rajeh1032 committed:
refactor: update loading state variables in CategoriesState and related components

ali-safwat commented on lib/Feature/AddressFeature/presentation/pages/saved_address_screen.dart:
```diff
@@ -0,0 +1,18 @@
+  Widget build(BuildContext context) {
+    return BlocProvider(
+        create: (context) =>
+            getIt<AddressViewModel>()..doIntent(GetAddressesEvent()),
```
you need to use gitit outside the build method

ali-safwat commented on lib/Feature/AddressFeature/presentation/widgets/saved_address_screen_body.dart:
```diff
@@ -0,0 +1,125 @@
+                  builder: (context, state) {
+                    return CustomElevatedButton(
+                      onPressed: () async {
+                        final result = await Navigator.of(context).push(
```
press is event so you need to make intent called need to navigate to screen x and view model emit go navigate or not 

don't make view do any action

ali-safwat reviewed (approved):
good work something need to enhance

rajeh1032 committed:
Merge branch 'Feature/Edit-Profile/FECA-30' of https://github.com/Eng-M0ataz/Flower-E-Commerce-App into Feature/Edit-Profile/FECA-30

Eng-M0ataz committed:
refactor : refactor in assets locL keys

rajeh1032 committed:
feat: enhance profile editing functionality with image selection and validation

rajeh1032 committed:
feat: enhance profile editing functionality with image selection and validation

Eng-M0ataz committed:
refactor: update import paths to use consistent casing for constants

Eng-M0ataz committed:
refactor: standardize import paths and clean up code formatting in sign-in and product details screens

Eng-M0ataz committed:
refactor: correct import paths for constants and improve code formatting across multiple files

rajeh1032 committed:
Merge branch 'Feature/Edit-Profile/FECA-30' of https://github.com/Eng-M0ataz/Flower-E-Commerce-App into Feature/Edit-Profile/FECA-30

rajeh1032 committed:
feat: update localization files and fix import paths for profile editing functionality

rajeh1032 committed:
Merge pull request #30 from Eng-M0ataz/Feature/Edit-Profile/FECA-30

feat/FECA-30 Edit Profile Business logic (API CALL) - Unit Test

Eng-M0ataz committed:
chore: update dependencies in pubspec.lock and clean up pubspec.yaml

- Added new dependencies: cross_file, file_selector (various platforms), flutter_plugin_android_lifecycle, and image_picker (various platforms).
- Removed duplicate entries for bloc_test and equatable in pubspec.yaml.
- Simplified MainLayout widget by consolidating screen management.
- Updated route generator to remove unnecessary parameters for MainLayout.

Eng-M0ataz committed:
feat: implement profile management features and update localization

- Added new SVG assets for profile editing functionality.
- Updated pubspec.yaml to correct output path for constants.
- Introduced new API services and data sources for profile management.
- Implemented use cases for editing profile and fetching logged user data.
- Enhanced profile screen with new view models and event handling.
- Updated localization files to include new keys for profile features.

Eng-M0ataz committed:
refactor: update import paths for execute_api in profile and categories data sources

rajeh1032 committed:
refactor: enhance address form validation and navigation flow

- Add validation for governorate, city, and recipient name fields
- Implement navigation events for add address flow using BlocConsumer
- Simplify form validation logic by removing custom error messages
- Add visual improvements to empty address list state with icon
- Clean up address state by removing unused AddressResponseEntity
- Add refresh functionality for address list updates

rajeh1032 committed:
refactor: enhance address form validation and navigation flow

- Add validation for governorate, city, and recipient name fields
- Implement navigation events for add address flow using BlocConsumer
- Simplify form validation logic by removing custom error messages
- Add visual improvements to empty address list state with icon
- Clean up address state by removing unused AddressResponseEntity
- Add refresh functionality for address list updates

rajeh1032 committed:
Merge branch 'Feature/Address-Details/FECA-44' of https://github.com/Eng-M0ataz/Flower-E-Commerce-App into Feature/Address-Details/FECA-44

rajeh1032 committed:
refactor: update import paths from Constants to Constantts and mainLayout/profile to editProfile

rajeh1032 committed:
Merge branch 'Feature/Address-Details/FECA-44' of https://github.com/Eng-M0ataz/Flower-E-Commerce-App into Feature/Address-Details/FECA-44
